src/rgw/rgw_sal_motr.cc

```diff
@@ -57,6 +61,168 @@ static std::string motr_global_indices[] = {
   RGW_MOTR_BUCKET_HD_IDX_NAME
 };
 
+struct MotrFidComparator {
+  bool operator()(const struct m0_fid& left, const struct m0_fid& right) const {
+    return left.f_container < right.f_container ||
+	   (left.f_container == right.f_container && left.f_key < right.f_key);
+  }
+};
+static map<struct m0_fid, MotrWatcher*, MotrFidComparator> motr_watchers;
+
+// process_fdmi_record() can't be implemented as MotrWatcher's member function
+// as the function pointer to a member funciton is a different type to fdmi's
+// callback function.
+// A workaround is to make process_fdmi_record() a static function in
+// the rgw::sal namespace. And it works as a dispatcher to further deliver
```

Comment:
I'd suggest putting "it works as a dispatcher..." in the beginning of the function documentation, and move the explanation about why it is not a class method below down under a note.